src/main/java/com/fasterxml/jackson/databind/deser/impl/PropertyBasedCreator.java

```diff
@@ -111,7 +111,12 @@ public static PropertyBasedCreator construct(DeserializationContext ctxt,
                 // 15-Apr-2020, tatu: [databind#962] Avoid getting deserializer for Inject-only
                 //     cases
                 if (!prop.isInjectionOnly()) {
-                    prop = prop.withValueDeserializer(ctxt.findContextualValueDeserializer(prop.getType(), prop));
+                    SettableBeanProperty existProperty = allProperties.find(prop.getName());
```

Comment:
In theory all Creator properties should have been added to `allProperties` earlier. But if not, that linkage should probably have been established earlier... at this point information in 2 is probably inconsistent.

Reply:
> As per CI, this breaks existing handling in a few unit tests so it wouldn't work as-is.
> I know there is something related to missing linking, but I am not sure it can be resolved by patch in this place in code.

I tried running tests locally with and without the changes from this PR.
There is this one test here (below image) fails.

![image](https://user-images.githubusercontent.com/61615301/216068548-6fbd5380-1380-40da-97fe-40bb3bcc5a5e.png)

- The reason for the test failure is that `prop.withValueDeserializer(ctxt.findContextualValueDeserializer(prop.getType(), prop));` returns an instance of `CreatorProperty`, while `allProperties.find(prop.getName());` returns `FieldProperty`
- `FieldProperty does not override `getCreatorIndex()` from `SetterableBeanProperty`

Reply:
Correct, `FieldProperty` does not have creator index since it does not represent a Creator (constructor, factory method) parameter.

So the change here is unfortunately wrong, I think; it tries to solve a real problem (Field and Creator properties not being linked, and annotations merged), but the solution should ideally work at an earlier point.

If my memory serves me, the problem occurs with auto-detecting Creator methods that are NOT explicitly annotated with `@JsonCreator` (or by having all parameters marked with `@JsonProperty`) -- those are created too late to link parameters within with possibly existing Field and MethodProperty (setter) instances.
What code here tries to do is to use earlier property, which may have more information: but that is not sufficient.

Reply:
> the problem occurs with auto-detecting Creator methods that are NOT explicitly annotated with @JsonCreator (or by having all parameters marked with @JsonProperty) -- those are created too late to link parameters within with possibly existing Field and MethodProperty (setter) instances.

Agreed. I was actually worried about this also.
Because implementation of `Auto-Detection`  between `FieldProperty` vs `CreatorProperty` seemed quite a work.

Reply:
In this version `2.13`  I feel like the clients lack control some sort of "Initialization priority".
So the real-problem of auto-detection you are referring to might need to be solved.

Since @jayjaehunchoi opened this PR to fix a bug from #3752 and might need solution, I think we need SOME of below?

1. Suggest work-around to solve the problem , until the real-problem is solved. Might also possibly be documented(At least for the documentation I can help 😅😅)

3.  some guidances to solve the real-problem from your perspective would help @cowtowncoder , maybe even create another issue for this real problem. (I am very interested in seeing how this goes)

5. documentation, warning for `jackson-databind` clients about this working.

Reply:
The real solution is to rewrite property (and creator) auto-detection. This has been something I'd like to do for years now. But I do not think that smaller work-arounds make sense; they just complicate code in the meantime.
So I agree that there is a problem, but I do not believe that local work-arounds help at this point: over time some have been added but at this point bigger change is needed.

This is actually what discussion #3719 is about.

Now, as to documentation: I am always pro-documentation, if I good place for its inclusion can be found.

Reply:
@jayjaehunchoi Would you like to close the PR if above conversation between me and @cowtowncoder meets your needs?

Reply:
sure